src/inference/src/cpp_interfaces/interface/ie_iinfer_request_internal.cpp

```diff
@@ -157,6 +158,139 @@ void IInferRequestInternal::SetBlob(const std::string& name, const Blob::Ptr& us
+                        "set_input_tensors/set_tensors error. Layout ",
+                        param->get_layout().to_string(),
+                        " is incorrect for operation with name '",
+                        name,
+                        "' with shape ",
+                        param->get_partial_shape());
+        auto batch = param->get_partial_shape()[batch_idx];
+
+        OPENVINO_ASSERT(batch.is_dynamic() || batch.get_length() == input_size,
+                        "set_input_tensors/set_tensors error. Input shape ",
+                        param->get_partial_shape(),
+                        "batch ",
+                        batch,
+                        "doesn't match with total blobs count: ",
+                        input_size);
+    }
+    OPENVINO_ASSERT(batch_idx == 0,
+                    "set_input_tensors/set_tensors is not currently supported for batch dimension index ",
+                    batch_idx,
+                    " != 0");
+    // Initial implementation always performs creation of new Blob and 'memcpy'
+    // In future consider checking if blobs point to contiguous range of memory
+    auto tmp_desc = blobs[0]->getTensorDesc();
+    tmp_desc.getDims()[batch_idx] = input_size;
+    auto blockingDims = tmp_desc.getBlockingDesc().getBlockDims();
+    blockingDims[batch_idx] = input_size;
+    auto blockingDesc = BlockingDesc(blockingDims, tmp_desc.getBlockingDesc().getOrder());
+    auto batched_desc = InferenceEngine::TensorDesc(tmp_desc.getPrecision(), tmp_desc.getDims(), blockingDesc);
+    auto desc_to_string = [](const TensorDesc& desc) {
+        std::stringstream s;
+        s << "{ " << desc.getLayout() << " " << desc.getPrecision().name();
+        s << "dim=(";
+        for (const auto& d : desc.getDims()) {
+            s << " " << d;
+        }
+        s << " ) }";
+        return s.str();
+    };
+    std::for_each(blobs.begin(), blobs.end(), [&batched_desc, &batch_idx, &desc_to_string](const Blob::Ptr& item) {
+        auto item_desc = item->getTensorDesc();
+        item_desc.getDims()[batch_idx] = batched_desc.getDims()[batch_idx];
+        OPENVINO_ASSERT(item_desc.getDims() == batched_desc.getDims() &&
+                            item_desc.getLayout() == batched_desc.getLayout() &&
+                            item_desc.getPrecision() == batched_desc.getPrecision() &&
+                            item_desc.getBlockingDesc().getOrder() == batched_desc.getBlockingDesc().getOrder(),
+                        "set_input_tensors/set_tensors error. Blob ",
+                        desc_to_string(item_desc),
+                        " is not compatible with batched blob ",
+                        desc_to_string(batched_desc));
+    });
+    _batched_blobs[name] = blobs;
+    _batched_descs[name] = batched_desc;
+
+    // Create batched blob. TODO: no API to get allocator of original blobs, use default one
```

Comment:
Still had to introduce new member 'batched_blobs'. Otherwise, if use inside 'inputs' - it will be overridden with resulting big blob, and next infer with new user's data will not work correctly. Test for this is also added